rust/agents/relayer/src/msg/metadata_builder.rs

```diff
@@ -92,4 +96,28 @@ impl MetadataBuilder {
             Ok(None)
         }
     }
+
+    async fn build_checkpoint_syncer(
+        &self,
+        validators: &[H256],
+    ) -> eyre::Result<MultisigCheckpointSyncer> {
+        let mut checkpoint_syncers: HashMap<H160, Arc<dyn CheckpointSyncer>> = HashMap::new();
+        let storage_locations = self
+            .validator_announce
+            .get_announced_storage_locations(validators)
```

Comment:
Is the plan to eventually cache this?
Noting that this is another RPC per attempt at message processing. But again unlikely to be a big issue now that we have backoffs for old messages

Reply:
Not concretely but probably we should cache it and then check back in if it looks like that storage location is no longer being written to